data/data-pipeline/data_pipeline/tests/test_score_calculation.py

```diff
@@ -0,0 +1,186 @@
+from ast import PyCF_TYPE_COMMENTS
+from pathlib import Path
+import pandas as pd
+import pytest
+import data_pipeline.score.field_names as field_names
+from ..score.score import Score
+from data_pipeline.utils import get_module_logger, score_folder_cleanup
+from data_pipeline.config import settings
+from data_pipeline.score.score_narwhal import ScoreNarwhal
+
+logger = get_module_logger(__name__)
+
+### TODO: we need to blow this out for all eight categories
+def _helper_test_percentile_against_threshold(
+    df, pctile_column, thresholded_column, threshold
+):
+    """Note - for the purpose of testing, this fills with False"""
+    minimum_flagged = (
+        df[df[thresholded_column].fillna(False)][pctile_column].min()
+        >= threshold
+    )
+
+    maximum_not_flagged = (
+        df[~df[thresholded_column].fillna(False)][pctile_column].max()
+        < threshold
+    )
+    assert (
+        minimum_flagged
+    ), f"For column {thresholded_column}, there is someone flagged below 90th percentile!"
+    assert (
+        maximum_not_flagged
+    ), f"For column {thresholded_column}, there is someone not flagged above 90th percentile!"
+    return True
+
+
+@pytest.fixture
+def final_score_df():
+    return pd.read_csv(
+        settings.APP_ROOT / "data" / "score" / "csv" / "full" / "usa.csv",
+        dtype={field_names.GEOID_TRACT_FIELD: str},
+    )
+
+
+@pytest.fixture
+def env_percentile_threshold():
+    return ScoreNarwhal(final_score_df).ENVIRONMENTAL_BURDEN_THRESHOLD
+
+
+# Climate factor thresholds
+@pytest.fixture
+def expected_pop_loss_percentile_columns():
+    return [
+        field_names.EXPECTED_POPULATION_LOSS_EXCEEDS_PCTILE_THRESHOLD,
+        field_names.EXPECTED_POPULATION_LOSS_RATE_FIELD
+        + field_names.PERCENTILE_FIELD_SUFFIX,
+    ]
+
+
+@pytest.fixture
+def expected_ag_loss_percentile_columns():
+    return [
+        field_names.EXPECTED_AGRICULTURAL_LOSS_EXCEEDS_PCTILE_THRESHOLD,
+        field_names.EXPECTED_AGRICULTURE_LOSS_RATE_FIELD
+        + field_names.PERCENTILE_FIELD_SUFFIX,
+    ]
+
+
+@pytest.fixture
+def expected_bldg_loss_percentile_columns():
+    return [
+        field_names.EXPECTED_BUILDING_LOSS_EXCEEDS_PCTILE_THRESHOLD,
+        field_names.EXPECTED_BUILDING_LOSS_RATE_FIELD
+        + field_names.PERCENTILE_FIELD_SUFFIX,
+    ]
+
+
+@pytest.fixture
+def expected_flood_percentile_columns():
+    return [
+        field_names.HIGH_FUTURE_FLOOD_RISK_FIELD,
+        field_names.FUTURE_FLOOD_RISK_FIELD
+        + field_names.PERCENTILE_FIELD_SUFFIX,
+    ]
+
+
+@pytest.fixture
+def expected_wildfire_percentile_columns():
+    return [
+        field_names.HIGH_FUTURE_WILDFIRE_RISK_FIELD,
+        field_names.FUTURE_WILDFIRE_RISK_FIELD
+        + field_names.PERCENTILE_FIELD_SUFFIX,
+    ]
+
+
+def test_expected_pop_loss_percentile_columns(
```

Comment:
this is really verbose, but the other option makes it much less modular. Thoughts?

Reply:
I agree, this is no fun. Given the structural similarity and the broad scope of these, I'd be pretty okay with a single "test_percentile_columns" that just iterates over these columns, given the clarity of the assertions. I can draft up a version after lunch for folks to think over.

Reply:
That sounds really great to me!